Mlem/Views/Tabs/Search/SearchResultListView.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  SearchResultListView.swift
+//  Mlem
+//
+//  Created by Sjmarf on 24/09/2023.
+//
+
+import SwiftUI
+
+struct SearchResultListView: View {
+    @EnvironmentObject var recentSearchesTracker: RecentSearchesTracker
+    @EnvironmentObject var contentTracker: ContentTracker<AnyContentModel>
+    
+    let showTypeLabel: Bool
+    
+    @State var shouldLoad = false
+    
+    var body: some View {
+        LazyVStack(spacing: 0) {
+            ForEach(contentTracker.items, id: \.uid) { contentModel in
+                Group {
+                    if let community = contentModel.wrappedValue as? CommunityModel {
+                        CommunityResultView(community: community, showTypeLabel: showTypeLabel)
+                    } else if let user = contentModel.wrappedValue as? UserModel {
+                        UserResultView(user: user, showTypeLabel: showTypeLabel)
+                    }
+                }
+                .simultaneousGesture(TapGesture().onEnded {
+                    recentSearchesTracker.addRecentSearch(contentModel)
+                    // Keyboard.dismiss()
```

Comment:
*nit* dead code?

Reply:
This was using a feature of SwiftUIX to dismiss the keyboard when you tap a result so that, when you navigate back to the search page, the search bar doesn't auto-focus. I started porting `Keyboard` over from SwiftUIX too, but it ended up looking like a lot of work so I thought I'd leave it out for now haha

I'll remove the comment 👍